Installing Nokogiri 1.6.2 fails on Mac with Xcode 5.1

As is well known, Xcode 5.1 includes a version of clang that is much pickier than previous versions. This affects Nokogiri:

```
linking shared-object nokogiri/nokogiri.bundle
clang: error: unknown argument: '-multiply_definedsuppress' [-Wunused-command-line-argument-hard-error-in-future]
clang: note: this will be a hard error (cannot be downgraded to a warning) in the future
make: *** [nokogiri.bundle] Error 1
```

By the way, preceding that there are also some warnings, but they are not fatal. I mention them here just in case it might be possible to fix them:

```
compiling xml_node.c
xml_node.c:17:14: warning: incompatible pointer types initializing 'xmlNodePtr' (aka 'struct _xmlNode *') with an expression of type 'struct _xmlDoc *' [-Wincompatible-pointer-types]
  xmlNodePtr doc = node->doc;
             ^     ~~~~~~~~~
xml_node.c:168:15: warning: 14 enumeration values not handled in switch: 'XML_ATTRIBUTE_NODE', 'XML_ENTITY_NODE', 'XML_DOCUMENT_NODE'... [-Wswitch]
      switch (reparentee->type) {
              ^
xml_node.c:186:15: warning: 15 enumeration values not handled in switch: 'XML_ATTRIBUTE_NODE', 'XML_ENTITY_NODE', 'XML_DOCUMENT_NODE'... [-Wswitch]
      switch (reparentee->type) {
              ^
xml_node.c:197:15: warning: 19 enumeration values not handled in switch: 'XML_ELEMENT_NODE', 'XML_ATTRIBUTE_NODE', 'XML_CDATA_SECTION_NODE'... [-Wswitch]
      switch (reparentee->type) {
              ^
xml_node.c:165:13: warning: 14 enumeration values not handled in switch: 'XML_CDATA_SECTION_NODE', 'XML_ENTITY_NODE', 'XML_PI_NODE'... [-Wswitch]
    switch (parent->type) {
            ^
5 warnings generated.
```

I was able to work around this by temporarily reverting to Xcode 5.0.2 (installation succeeds with no warnings), but that is not a viable long-term solution.

--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -311,6 +311,9 @@ when 'mingw32', /mswin/
   $CFLAGS << " -DXP_WIN -DXP_WIN32 -DUSE_INCLUDED_VASPRINTF"
 when /solaris/
   $CFLAGS << " -DUSE_INCLUDED_VASPRINTF"
+when /darwin/
+  # TODO: maybe make it stricter to only work on stock ruby? How to tell?
+  $CFLAGS << " -Wno-error=unused-command-line-argument-hard-error-in-future"
 else
   $CFLAGS << " -g -DXP_UNIX"
 end

